web_poet/pages.py

```diff
@@ -137,3 +150,20 @@ class WebPage(ItemPage[ItemT], ResponseShortcutsMixin):
 
 
 ItemWebPage = _create_deprecated_class("ItemWebPage", WebPage, warn_once=False)
+
+
+@attr.s(auto_attribs=True)
+class SelectorExtractor(Extractor[ItemT]):
+    """Extractor that takes a :class:`parsel.Selector` and provides
+    XPath / CSS shortcuts.
+    """
+
+    selector: parsel.Selector
+
+    def xpath(self, query, **kwargs) -> parsel.SelectorList:
```

Comment:
This code is exactly the same as in SelectableMixin. What do you think about creating CssXpathMixin, and using it both here and in SelectableMixin?

Reply:
"Shortcuts are copied from SelectableMixin, I don't think we need a separate mixin for those but it's possible :)" was about this :)

Reply:
Yeah, let's do it :) The code is simple, but still, there is query and **kwargs, and typing, it seems it's worth it to avoid duplication in this case.

Reply:
This isn't as straightforward as I thought as `selector` is a property in `SelectableMixin` but an attribute in `SelectorExtractor` so I'll look into it later (unless we want to just silence mypy on this).

Reply:
I went with silencing mypy because even if we find a clean way to declare `selector: parsel.Selector` in `CssXpathMixin` we still need to redeclare it in `SelectorExtractor` because of `attrs` (unless there is a way around that too).